src/main/java/org/openrewrite/java/migrate/lang/var/UseVarForGenericMethodInvocations.java

```diff
@@ -81,17 +80,26 @@ public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations v
             return transformToVar(vd, new ArrayList<>(), new ArrayList<>());
         }
 
+        private static boolean allArgumentsEmpty(J.MethodInvocation invocation){
+            for (Expression argument : invocation.getArguments()) {
+                if (!(argument instanceof J.Empty)) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
         private J.VariableDeclarations transformToVar(J.VariableDeclarations vd, List<JavaType> leftTypes, List<JavaType> rightTypes) {
             Expression initializer = vd.getVariables().get(0).getInitializer();
             String simpleName = vd.getVariables().get(0).getSimpleName();
 
             // if left is defined but not right, copy types to initializer
             if(rightTypes.isEmpty() && !leftTypes.isEmpty()) {
                 // we need to switch type infos from left to right here
-                List<Expression> typeArgument = leftTypes.stream()
-                        .map(t ->
-                                new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, ((JavaType.Class)t).getClassName(), t, null))
-                        .collect(Collectors.toList());
+                List<Expression> typeArgument = new ArrayList<>();
+                for (JavaType t : leftTypes) {
+                    typeArgument.add(new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, ((JavaType.Class) t).getClassName(), t, null));
+                }
```

Comment:
I forgot to mention the use of `ListUtils`; that stops us creating a new list if it's not necessary, so a neat addition here (and conforms to other recipes).
```suggestion
                List<Expression> typeArgument = ListUtils.map(leftTypes, t -> {
                    typeArgument.add(new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, ((JavaType.Class) t).getClassName(), t, null));
                });
```

Reply:
Nevermind; read this wrong!